# Introduction

In this notebook, we will do a comprehensive analysis of the Android app market by comparing thousands of apps in the Google Play store.

# About the Dataset of Google Play Store Apps & Reviews

**Data Source:** <br>
App and review data was scraped from the Google Play Store by Lavanya Gupta in 2018 from Kaggle. Original files listed [here](
https://www.kaggle.com/lava18/google-play-store-apps).


# Import Statements

In [1]:
import pandas as pd
import plotly.express as px

# Notebook Presentation

In [2]:
# Show numeric output in decimal format e.g., 2.15
pd.options.display.float_format = '{:,.2f}'.format

# Reading the Dataset

In [3]:
df_apps = pd.read_csv('apps.csv')

# Data Cleaning

In [4]:
print(f"Shape: {df_apps.shape}")
df_apps.head()

Shape: (10841, 12)


App         Category  Rating  \
0                       Ak Parti Yardım Toplama           SOCIAL     NaN   
1                    Ain Arabic Kids Alif Ba ta           FAMILY     NaN   
2  Popsicle Launcher for Android P 9.0 launcher  PERSONALIZATION     NaN   
3                     Command & Conquer: Rivals           FAMILY     NaN   
4                                    CX Network         BUSINESS     NaN   

   Reviews  Size_MBs Installs  Type   Price Content_Rating           Genres  \
0        0      8.70        0  Paid  $13.99           Teen           Social   
1        0     33.00        0  Paid   $2.99       Everyone        Education   
2        0      5.50        0  Paid   $1.49       Everyone  Personalization   
3        0     19.00        0   NaN       0   Everyone 10+         Strategy   
4        0     10.00        0  Free       0       Everyone         Business   

     Last_Updated         Android_Ver  
0   July 28, 2017          4.1 and up  
1  April 15, 2016          3.0 and up  
2   July 11, 2018          4.2 and up  
3   June 28, 2018  Varies with device  
4  August 6, 2018          4.1 and up

In [5]:
df_apps.sample(5)

App            Category  Rating  Reviews  \
9367                Smart Launcher 5     PERSONALIZATION    4.40   512102   
8893        Babbel – Learn Languages           EDUCATION    4.30   266935   
9955  Hustle Castle: Fantasy Kingdom                GAME    4.70   584126   
8052              CrossFire: Legends                GAME    4.30   144040   
4899        Calorie Counter - Macros  HEALTH_AND_FITNESS    4.00     3061   

      Size_MBs    Installs  Type Price Content_Rating            Genres  \
9367      6.15  10,000,000  Free     0       Everyone   Personalization   
8893     21.00  10,000,000  Free     0       Everyone         Education   
9955     64.00  10,000,000  Free     0   Everyone 10+      Role Playing   
8052     29.00   5,000,000  Free     0           Teen            Action   
4899      5.50     100,000  Free     0       Everyone  Health & Fitness   

         Last_Updated         Android_Ver  
9367    July 16, 2018  Varies with device  
8893    July 30, 2018          4.4 and up  
9955    July 27, 2018          4.1 and up  
8052    July 25, 2018          4.0 and up  
4899  August 27, 2016          5.0 and up

### Drop Unused Columns

Let's remove unnecessary columns: `Last_Updated` and `Android_Version`

In [6]:
df_apps.drop(['Last_Updated', 'Android_Ver'], axis=1, inplace=True)
df_apps.head()

App         Category  Rating  \
0                       Ak Parti Yardım Toplama           SOCIAL     NaN   
1                    Ain Arabic Kids Alif Ba ta           FAMILY     NaN   
2  Popsicle Launcher for Android P 9.0 launcher  PERSONALIZATION     NaN   
3                     Command & Conquer: Rivals           FAMILY     NaN   
4                                    CX Network         BUSINESS     NaN   

   Reviews  Size_MBs Installs  Type   Price Content_Rating           Genres  
0        0      8.70        0  Paid  $13.99           Teen           Social  
1        0     33.00        0  Paid   $2.99       Everyone        Education  
2        0      5.50        0  Paid   $1.49       Everyone  Personalization  
3        0     19.00        0   NaN       0   Everyone 10+         Strategy  
4        0     10.00        0  Free       0       Everyone         Business

### Removing NaN values in Ratings

In [7]:
nan_rows = df_apps[df_apps.Rating.isna()]
print(nan_rows.shape)
nan_rows.head()

(1474, 10)


App         Category  Rating  \
0                       Ak Parti Yardım Toplama           SOCIAL     NaN   
1                    Ain Arabic Kids Alif Ba ta           FAMILY     NaN   
2  Popsicle Launcher for Android P 9.0 launcher  PERSONALIZATION     NaN   
3                     Command & Conquer: Rivals           FAMILY     NaN   
4                                    CX Network         BUSINESS     NaN   

   Reviews  Size_MBs Installs  Type   Price Content_Rating           Genres  
0        0      8.70        0  Paid  $13.99           Teen           Social  
1        0     33.00        0  Paid   $2.99       Everyone        Education  
2        0      5.50        0  Paid   $1.49       Everyone  Personalization  
3        0     19.00        0   NaN       0   Everyone 10+         Strategy  
4        0     10.00        0  Free       0       Everyone         Business

In [8]:
df_apps_clean = df_apps.dropna()
df_apps_clean.shape

(9367, 10)

### Removing Duplicates

In [9]:
duplicated_rows = df_apps_clean[df_apps_clean.duplicated()]
print(duplicated_rows.shape)
duplicated_rows.head()

(476, 10)


App Category  Rating  Reviews  \
946                       420 BZ Budeze Delivery  MEDICAL    5.00        2   
1133                                 MouseMingle   DATING    2.70        3   
1196  Cardiac diagnosis (heart rate, arrhythmia)  MEDICAL    4.40        8   
1231                                Sway Medical  MEDICAL    5.00        3   
1247              Chat Kids - Chat Room For Kids   DATING    4.70        6   

      Size_MBs Installs  Type   Price Content_Rating   Genres  
946      11.00      100  Free       0     Mature 17+  Medical  
1133      3.90      100  Free       0     Mature 17+   Dating  
1196      6.50      100  Paid  $12.99       Everyone  Medical  
1231     22.00      100  Free       0       Everyone  Medical  
1247      4.90      100  Free       0     Mature 17+   Dating

In [10]:
df_apps_clean[df_apps_clean.App == 'Instagram']

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   
10808  Instagram   SOCIAL    4.50  66577446      5.30  1,000,000,000  Free   
10809  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   
10810  Instagram   SOCIAL    4.50  66509917      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social  
10808     0           Teen  Social  
10809     0           Teen  Social  
10810     0           Teen  Social

In [11]:
df_apps_clean = df_apps_clean.drop_duplicates()
df_apps_clean[df_apps_clean.App == 'Instagram']

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   
10808  Instagram   SOCIAL    4.50  66577446      5.30  1,000,000,000  Free   
10810  Instagram   SOCIAL    4.50  66509917      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social  
10808     0           Teen  Social  
10810     0           Teen  Social

In [12]:
df_apps_clean[df_apps_clean.App == 'Instagram']

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   
10808  Instagram   SOCIAL    4.50  66577446      5.30  1,000,000,000  Free   
10810  Instagram   SOCIAL    4.50  66509917      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social  
10808     0           Teen  Social  
10810     0           Teen  Social

In [13]:
# need to specify the subset for indentifying duplicates
df_apps_clean = df_apps_clean.drop_duplicates(subset=['App', 'Type', 'Price'])
df_apps_clean[df_apps_clean.App == 'Instagram']

App Category  Rating   Reviews  Size_MBs       Installs  Type  \
10806  Instagram   SOCIAL    4.50  66577313      5.30  1,000,000,000  Free   

      Price Content_Rating  Genres  
10806     0           Teen  Social

In [14]:
df_apps_clean.shape

(8199, 10)

# Highest Rated Apps

In [15]:
df_apps_clean.sort_values('Rating', ascending=False).head()

App     Category  Rating  Reviews  Size_MBs Installs  \
21    KBA-EZ Health Guide      MEDICAL    5.00        4     25.00        1   
1230         Sway Medical      MEDICAL    5.00        3     22.00      100   
1227    AJ Men's Grooming    LIFESTYLE    5.00        2     22.00      100   
1224       FK Dedinje BGD       SPORTS    5.00       36      2.60      100   
1223      CB VIDEO VISION  PHOTOGRAPHY    5.00       13      2.60      100   

      Type Price Content_Rating       Genres  
21    Free     0       Everyone      Medical  
1230  Free     0       Everyone      Medical  
1227  Free     0       Everyone    Lifestyle  
1224  Free     0       Everyone       Sports  
1223  Free     0       Everyone  Photography

Only apps with very few reviews (and a low number on installs) have perfect 5 star ratings (most likely by friends and family).

# 5 Largest Apps in terms of Size (MBs)

In [16]:
df_apps_clean.sort_values('Size_MBs', ascending=False).head()

App            Category  Rating  Reviews  \
9942   Talking Babsy Baby: Baby Games           LIFESTYLE    4.00   140995   
10687          Hungry Shark Evolution                GAME    4.50  6074334   
9943            Miami crime simulator                GAME    4.00   254518   
9944     Gangster Town: Vice District              FAMILY    4.30    65146   
3144                       Vi Trainer  HEALTH_AND_FITNESS    3.60      124   

       Size_MBs     Installs  Type Price Content_Rating  \
9942     100.00   10,000,000  Free     0       Everyone   
10687    100.00  100,000,000  Free     0           Teen   
9943     100.00   10,000,000  Free     0     Mature 17+   
9944     100.00   10,000,000  Free     0     Mature 17+   
3144     100.00        5,000  Free     0       Everyone   

                       Genres  
9942   Lifestyle;Pretend Play  
10687                  Arcade  
9943                   Action  
9944               Simulation  
3144         Health & Fitness

Here we can clearly see that there seems to be an upper bound of 100 MB for the size of an app. A quick google search would also have revealed that this limit is imposed by the Google Play Store itself.

# 5 App with Most Reviews

In [17]:
apps_50_most_reviews = df_apps_clean.sort_values('Reviews', ascending=False).head(50)
apps_50_most_reviews.head()

App       Category  Rating  \
10805                                  Facebook         SOCIAL    4.10   
10785                        WhatsApp Messenger  COMMUNICATION    4.40   
10806                                 Instagram         SOCIAL    4.50   
10784  Messenger – Text and Video Chat for Free  COMMUNICATION    4.00   
10650                            Clash of Clans           GAME    4.60   

        Reviews  Size_MBs       Installs  Type Price Content_Rating  \
10805  78158306      5.30  1,000,000,000  Free     0           Teen   
10785  69119316      3.50  1,000,000,000  Free     0       Everyone   
10806  66577313      5.30  1,000,000,000  Free     0           Teen   
10784  56642847      3.50  1,000,000,000  Free     0       Everyone   
10650  44891723     98.00    100,000,000  Free     0   Everyone 10+   

              Genres  
10805         Social  
10785  Communication  
10806         Social  
10784  Communication  
10650       Strategy

In [18]:
apps_50_most_reviews[apps_50_most_reviews["Type"] == "Paid"].values.any()

False

If we look at the number of reviews, we can find the most popular apps on the Android App Store. These include the usual suspects: Facebook, WhatsApp, Instagram etc. What’s also notable is that the list of the top 50 most reviewed apps does not include a single paid app!



# Visualizing Categorical Data: Content Ratings

In [19]:
ratings = df_apps_clean.Content_Rating.value_counts()
ratings

Content_Rating
Everyone           6621
Teen                912
Mature 17+          357
Everyone 10+        305
Adults only 18+       3
Unrated               1
Name: count, dtype: int64

In [20]:
fig = px.pie(labels=ratings.index, 
             values=ratings.values,
             title="Content Rating",
             names=ratings.index,
)
fig.update_traces(textposition='outside', textinfo='percent+label')

fig.show()

# Apps with most Number of Installs

In [21]:
df_apps_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8199 entries, 21 to 10835
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             8199 non-null   object 
 1   Category        8199 non-null   object 
 2   Rating          8199 non-null   float64
 3   Reviews         8199 non-null   int64  
 4   Size_MBs        8199 non-null   float64
 5   Installs        8199 non-null   object 
 6   Type            8199 non-null   object 
 7   Price           8199 non-null   object 
 8   Content_Rating  8199 non-null   object 
 9   Genres          8199 non-null   object 
dtypes: float64(2), int64(1), object(7)
memory usage: 704.6+ KB


In [22]:
df_apps_clean[['App', 'Installs']].groupby('Installs').count()

App
Installs           
1                 3
1,000           698
1,000,000      1417
1,000,000,000    20
10               69
10,000          988
10,000,000      933
100             303
100,000        1096
100,000,000     189
5                 9
5,000           425
5,000,000       607
50               56
50,000          457
50,000,000      202
500             199
500,000         504
500,000,000      24

In [23]:
df_apps_clean.Installs = df_apps_clean.Installs.astype(str).str.replace(',', "")

df_apps_clean.Installs = pd.to_numeric(df_apps_clean.Installs)

df_apps_clean[['App', 'Installs']].groupby('Installs').count()

App
Installs        
1              3
5              9
10            69
50            56
100          303
500          199
1000         698
5000         425
10000        988
50000        457
100000      1096
500000       504
1000000     1417
5000000      607
10000000     933
50000000     202
100000000    189
500000000     24
1000000000    20

# The Most Expensive Apps and Calculating a (ballpark) Sales Revenue Estimate

Let's examine the Price column more closely.


In [24]:
df_apps_clean.Price = df_apps_clean.Price.astype(str).str.replace('$', "")
df_apps_clean.Price = pd.to_numeric(df_apps_clean.Price)

df_apps_clean.sort_values('Price', ascending=False).head(20)

App   Category  Rating  Reviews  Size_MBs  \
3946        I'm Rich - Trump Edition  LIFESTYLE    3.60      275      7.30   
2461              I AM RICH PRO PLUS    FINANCE    4.00       36     41.00   
4606               I Am Rich Premium    FINANCE    4.10     1867      4.70   
3145              I am rich(premium)    FINANCE    3.50      472      0.94   
3554                      💎 I'm rich  LIFESTYLE    3.80      718     26.00   
5765                       I am rich  LIFESTYLE    3.80     3547      1.80   
1946  I am rich (Most expensive app)    FINANCE    4.10      129      2.70   
2775                   I Am Rich Pro     FAMILY    4.40      201      2.70   
3221                  I am Rich Plus     FAMILY    4.00      856      8.70   
3114                       I am Rich    FINANCE    4.30      180      3.80   
1331          most expensive app (H)     FAMILY    4.30        6      1.50   
2394                      I am Rich!    FINANCE    3.80       93     22.00   
3897                       I Am Rich     FAMILY    3.60      217      4.90   
2193             I am extremely Rich  LIFESTYLE    2.90       41      2.90   
3856                   I am rich VIP  LIFESTYLE    3.80      411      2.60   
2281       Vargo Anesthesia Mega App    MEDICAL    4.60       92     32.00   
1407                    LTC AS Legal    MEDICAL    4.00        6      1.30   
2629                I am Rich Person  LIFESTYLE    4.20      134      1.80   
2481         A Manual of Acupuncture    MEDICAL    3.50      214     68.00   
4264         Golfshot Plus: Golf GPS     SPORTS    4.10     3387     25.00   

      Installs  Type  Price Content_Rating         Genres  
3946     10000  Paid 400.00       Everyone      Lifestyle  
2461      1000  Paid 399.99       Everyone        Finance  
4606     50000  Paid 399.99       Everyone        Finance  
3145      5000  Paid 399.99       Everyone        Finance  
3554     10000  Paid 399.99       Everyone      Lifestyle  
5765    100000  Paid 399.99       Everyone      Lifestyle  
1946      1000  Paid 399.99           Teen        Finance  
2775      5000  Paid 399.99       Everyone  Entertainment  
3221     10000  Paid 399.99       Everyone  Entertainment  
3114      5000  Paid 399.99       Everyone        Finance  
1331       100  Paid 399.99       Everyone  Entertainment  
2394      1000  Paid 399.99       Everyone        Finance  
3897     10000  Paid 389.99       Everyone  Entertainment  
2193      1000  Paid 379.99       Everyone      Lifestyle  
3856     10000  Paid 299.99       Everyone      Lifestyle  
2281      1000  Paid  79.99       Everyone        Medical  
1407       100  Paid  39.99       Everyone        Medical  
2629      1000  Paid  37.99       Everyone      Lifestyle  
2481      1000  Paid  33.99       Everyone        Medical  
4264     50000  Paid  29.99       Everyone         Sports

What’s going on here? There are 15 I am Rich Apps in the Google Play Store apparently. They all cost $300 or more, which is the main point of the app. The story goes that in 2008, Armin Heinrich released the very first I am Rich app in the iOS App Store for $999.90. The app does absolutely nothing. It just displays the picture of a gemstone and can be used to prove to your friends how rich you are. Armin actually made a total of 7 sales before the app was hastily removed by Apple. Nonetheless, it inspired a bunch of copycats on the Android App Store, but if you search today, you’ll find all of these apps have disappeared as well. The high installation numbers are likely gamed by making the app was available for free at some point to get reviews and appear more legitimate.



### The most expensive apps sub $250

In [25]:
df_apps_clean = df_apps_clean[df_apps_clean['Price'] < 250]
df_apps_clean.sort_values('Price', ascending=False).head(5)

App   Category  Rating  Reviews  Size_MBs  \
2281  Vargo Anesthesia Mega App    MEDICAL    4.60       92     32.00   
1407               LTC AS Legal    MEDICAL    4.00        6      1.30   
2629           I am Rich Person  LIFESTYLE    4.20      134      1.80   
2481    A Manual of Acupuncture    MEDICAL    3.50      214     68.00   
2463         PTA Content Master    MEDICAL    4.20       64     41.00   

      Installs  Type  Price Content_Rating     Genres  
2281      1000  Paid  79.99       Everyone    Medical  
1407       100  Paid  39.99       Everyone    Medical  
2629      1000  Paid  37.99       Everyone  Lifestyle  
2481      1000  Paid  33.99       Everyone    Medical  
2463      1000  Paid  29.99       Everyone    Medical

When we look at the top 5 apps now, we see that 4 out of 5 are medical apps.

### Highest Grossing Paid Apps (ballpark estimate)
Let's add a column called 'Revenue_Estimate' to the DataFrame.

In [26]:
df_apps_clean['Revenue_Estimate'] = df_apps_clean.Installs.mul(df_apps_clean.Price)
df_apps_clean.sort_values('Revenue_Estimate', ascending=False)[:10]

App     Category  Rating  Reviews  Size_MBs  \
9220                      Minecraft       FAMILY    4.50  2376564     19.00   
8825                  Hitman Sniper         GAME    4.60   408292     29.00   
7151  Grand Theft Auto: San Andreas         GAME    4.40   348962     26.00   
7477            Facetune - For Free  PHOTOGRAPHY    4.40    49553     48.00   
7977        Sleep as Android Unlock    LIFESTYLE    4.50    23966      0.85   
6594            DraStic DS Emulator         GAME    4.60    87766     12.00   
6082                   Weather Live      WEATHER    4.50    76593      4.75   
7954                    Bloons TD 5       FAMILY    4.60   190086     94.00   
7633        Five Nights at Freddy's         GAME    4.60   100805     50.00   
6746     Card Wars - Adventure Time       FAMILY    4.30   129603     23.00   

      Installs  Type  Price Content_Rating                     Genres  \
9220  10000000  Paid   6.99   Everyone 10+  Arcade;Action & Adventure   
8825  10000000  Paid   0.99     Mature 17+                     Action   
7151   1000000  Paid   6.99     Mature 17+                     Action   
7477   1000000  Paid   5.99       Everyone                Photography   
7977   1000000  Paid   5.99       Everyone                  Lifestyle   
6594   1000000  Paid   4.99       Everyone                     Action   
6082    500000  Paid   5.99       Everyone                    Weather   
7954   1000000  Paid   2.99       Everyone                   Strategy   
7633   1000000  Paid   2.99           Teen                     Action   
6746   1000000  Paid   2.99   Everyone 10+    Card;Action & Adventure   

      Revenue_Estimate  
9220     69,900,000.00  
8825      9,900,000.00  
7151      6,990,000.00  
7477      5,990,000.00  
7977      5,990,000.00  
6594      4,990,000.00  
6082      2,995,000.00  
7954      2,990,000.00  
7633      2,990,000.00  
6746      2,990,000.00

This generously assumes of course that all the installs would have been made at the listed price, which is unlikely, as there are always promotions and free give-aways on the App Stores.

The top spot of the highest-grossing paid app goes to … Minecraft at close to $70 million. It’s quite interesting that Minecraft (along with Bloons and Card Wars) is actually listed in the Family category rather than in the Game category. If we include these titles, we see that 7 out the top 10 highest-grossing apps are games. The Google Play Store seems to be quite flexible with its category labels.

# Analysing App Categories

Let’s analyse this with bar charts and scatter plots and figure out which categories are dominating the market.



In [27]:
print(f"Number of categories: {df_apps_clean.Category.nunique()}")
top10_category = df_apps_clean.Category.value_counts()[:10]
top10_category

Number of categories: 33


Category
FAMILY             1606
GAME                910
TOOLS               719
PRODUCTIVITY        301
PERSONALIZATION     298
LIFESTYLE           297
FINANCE             296
MEDICAL             292
PHOTOGRAPHY         263
BUSINESS            262
Name: count, dtype: int64

### Highest Competition (Number of Apps) - Vertical Bar Chart

In [28]:
bar = px.bar(
        x = top10_category.index, # index = category name
        y = top10_category.values,
        title="Highest Competition (Number of Apps in a category)")

bar.update_layout(xaxis_title='Number of Apps', yaxis_title='Category')

bar.show()

Based on the number of apps, the Family and Game categories are the most competitive. Releasing yet another app into these categories will make it hard to get noticed.

### Most Popular Categories (Highest Downloads) - Horizontal Bar Chart

In [29]:
category_installs = df_apps_clean.groupby('Category').agg({'Installs': pd.Series.sum})
category_installs.sort_values('Installs', ascending=True, inplace=True)

In [30]:
h_bar = px.bar(
        x = category_installs.Installs,
        y = category_installs.index,
        orientation='h',
        title='Category Popularity')

h_bar.update_layout(xaxis_title='Number of Downloads', yaxis_title='Category')
h_bar.show()

Now we see that Games and Tools are actually the most popular categories. If we plot the popularity of a category next to the number of apps in that category we can get an idea of how concentrated a category is. Do few apps have most of the downloads or are the downloads spread out over many apps?

### Category Concentration - Downloads vs. Competition

In [31]:
cat_number = df_apps_clean.groupby('Category').agg({'App': pd.Series.count})
cat_merged_df = pd.merge(cat_number, category_installs, on='Category', how="inner")
print(f'The dimensions of the DataFrame are: {cat_merged_df.shape}')
cat_merged_df.sort_values('Installs', ascending=False)[:10]

The dimensions of the DataFrame are: (33, 2)


App     Installs
Category                             
GAME                 910  13858762717
COMMUNICATION        257  11039241530
TOOLS                719   8099724500
PRODUCTIVITY         301   5788070180
SOCIAL               203   5487841475
PHOTOGRAPHY          263   4649143130
FAMILY              1606   4437554490
VIDEO_PLAYERS        148   3916897200
TRAVEL_AND_LOCAL     187   2894859300
NEWS_AND_MAGAZINES   204   2369110650

In [32]:
scatter = px.scatter(cat_merged_df, # data
                     x='App', # column name
                     y='Installs',
                     title='Category Concentration',
                     size='App',
                     hover_name=cat_merged_df.index,
                     color='Installs'
) 

scatter.update_layout(xaxis_title="Number of Apps (Lower=More Concentrated)",
                      yaxis_title="Installs",
                      yaxis=dict(type='log'))

scatter.show()

What we see is that the categories like Family, Tools, and Game have many different apps sharing a high number of downloads. But for the categories like video players and entertainment, all the downloads are concentrated in very few apps.

# Competition in Genres - Colour Scales

### Extracting Nested Data from a Column

Here we see that the semi-colon ``;`` separates the genre names.

In [33]:
print(f"Number of Genres: {len(df_apps_clean.Genres.unique())}")
df_apps_clean.Genres.value_counts().sort_values(ascending=True)[:5]

Number of Genres: 114


Genres
Lifestyle;Pretend Play      1
Strategy;Education          1
Adventure;Education         1
Role Playing;Brain Games    1
Tools;Education             1
Name: count, dtype: int64

In [34]:
# Spliting the strings on the semi-colon and then .stacking them. 
stack = df_apps_clean.Genres.str.split(';', expand=True).stack()
print(f'We now have a single column with shape: {stack.shape}')
num_genres = stack.value_counts()
print(f'Number of genres: {len(num_genres)}')

We now have a single column with shape: (8564,)
Number of genres: 53


In [35]:
bar = px.bar(
        x = num_genres.index[:15],
        y = num_genres.values[:15],
        title='Top Genres',
        hover_name=num_genres.index[:15],
        color=num_genres.values[:15],
        color_continuous_scale='Agsunset'
)

bar.update_layout(xaxis_title='Genre',
                  yaxis_title='Number of Apps',
                  coloraxis_showscale=False)

bar.show()

# Free vs. Paid Apps per Category - Grouped Bar Charts

In [36]:
df_apps_clean.Type.value_counts()

Type
Free    7595
Paid     589
Name: count, dtype: int64

In [37]:
df_free_vs_paid = df_apps_clean.groupby(["Category", "Type"], 
                                      as_index=False).agg({'App': pd.Series.count})
df_free_vs_paid.sort_values('App')

Category  Type   App
3    AUTO_AND_VEHICLES  Paid     1
24      FOOD_AND_DRINK  Paid     2
38  NEWS_AND_MAGAZINES  Paid     2
40           PARENTING  Paid     2
17       ENTERTAINMENT  Paid     2
..                 ...   ...   ...
31           LIFESTYLE  Free   284
21             FINANCE  Free   289
53               TOOLS  Free   656
25                GAME  Free   834
19              FAMILY  Free  1456

[61 rows x 3 columns]

In [38]:
g_bar = px.bar(df_free_vs_paid, 
               x='Category', 
               y='App',
               title='Free vs Paid Apps by Category',
               color='Type', 
               barmode='group',)

g_bar.update_layout(xaxis_title='Category',
                    yaxis_title='Number of Apps',
                    xaxis={'categoryorder':'total descending'},
                    yaxis=dict(type='log'),
                    )

g_bar.show()

What we see is that while there are very few paid apps on the Google Play Store, some categories have relatively more paid apps than others, including Personalization, Medical and Weather. So, depending on the category you are targeting, it might make sense to release a paid-for app.

# Lost Downloads for Paid Apps - Box Plots: 

Lets ask ourselves some questions:

1. How much should you charge? What are other apps charging in that category?

2. How much revenue could you make?

3. And how many downloads are you potentially giving up because your app is paid?

In [39]:
box = px.box(df_apps_clean, 
             y='Installs',
             x='Type',
             color='Type',
             notched=True,
             points='all',
             title='How Many Downloads are Paid Apps Giving Up?'
)

box.update_layout(yaxis=dict(type='log'))

box.show()


From the hover text in the chart, we see that the median number of downloads for free apps is 500,000, while the median number of downloads for paid apps is around 5,000! This is massively lower.<br>
But does this mean we should give up on selling a paid app? Let’s see how much revenue we would estimate per category.

# Revenue by App Category - Box Plots: 

In [40]:
df_paid_apps = df_apps_clean[df_apps_clean['Type'] == 'Paid']

In [41]:
box = px.box(df_paid_apps, 
             x='Category', 
             y='Revenue_Estimate',
             title='How Much Can Paid Apps Earn?')

box.update_layout(xaxis_title='Category',
                  yaxis_title='Paid App Ballpark Revenue',
                  xaxis={'categoryorder':'min ascending'},
                  yaxis=dict(type='log'))


box.show()

If an Android app costs $30,000 to develop, then the average app in very few categories would cover that development cost. The median paid photography app earned about $20,000. Many more app’s revenues were even lower - meaning they would need other sources of revenue like advertising or in-app purchases to make up for their development costs. However, certain app categories seem to contain a large number of outliers that have much higher (estimated) revenue - for example in Medical, Personalisation, Tools, Game, and Family.

# How Much Can You Charge? Examine Paid App Pricing Strategies by Category

So, if you were to list a paid app, how should you price it? To help you decide we can look at how your competitors in the same category price their apps.

In [42]:
df_paid_apps.Price.median()

2.99

In [43]:
box = px.box(df_paid_apps, 
             x='Category', 
             y="Price",
             title='Price per Category')

box.update_layout(xaxis_title='Category',
                  yaxis_title='Paid App Price',
                  xaxis={'categoryorder':'max descending'},
                  yaxis=dict(type='log'))


box.show()

The median price for an Android app is $2.99.

However, some categories have higher median prices than others. This time we see that Medical apps have the most expensive apps as well as a median price of $5.49. In contrast, Personalisation apps are quite cheap on average at $1.49. Other categories which higher median prices are Business ($4.99) and Dating ($6.99). It seems like customers who shop in these categories are not so concerned about paying a bit extra for their apps.